File: 作业13-求最大面积/java/Solution.java
public class Solution {
	public static int maxArea(int[] nums) {
		int maxArea = 0,left = 0, right = nums.length - 1;
		while (left < right) {
			maxArea = Math.max(maxArea, Math.min(nums[left], nums[right])  * (right - left));
			if (nums[left] > nums[right]) {
				right--;
			} else {
				left++;
			}
		}

		return maxArea;
	}
}

File: 作业13-求最大面积/php/Solution.php
<?php

class Solution {
	function maxArea($nums) {
		$maxArea = 0; $left = 0; $right = count($nums) - 1;
		while ($left < $right) {
			$maxArea = max($maxArea, min($nums[$left], $nums[$right]) * ($right - $left));
			if ($nums[$left] > $nums[$right]) {
				$right--;
			} else {
				$left++;
			}
		}
		return $maxArea;
	}
}